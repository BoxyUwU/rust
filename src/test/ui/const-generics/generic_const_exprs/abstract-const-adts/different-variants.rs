#![feature(generic_const_exprs, adt_const_params)]
#![allow(incomplete_features)]

#[derive(PartialEq, Eq)]
enum MyEnum {
    Var1 { field: u32 },
    Var2 { field: u32 },
}

struct HasMyEnum<const N: MyEnum>;

fn foo<const N: u32>(a: HasMyEnum<{ MyEnum::Var1 { field: N } }>) {
    bar::<{ MyEnum::Var1 { field: N } }>();
    bar::<{ MyEnum::Var2 { field: N } }>();
    //~^ error: unconstrained generic constant
}

fn bar<const N: MyEnum>() {}

fn main() {}
