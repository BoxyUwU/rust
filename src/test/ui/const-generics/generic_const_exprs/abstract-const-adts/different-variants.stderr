error: unconstrained generic constant
  --> $DIR/different-variants.rs:14:11
   |
LL |     bar::<{ MyEnum::Var2 { field: N } }>();
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: try adding a `where` bound using this expression: `where [(); { MyEnum::Var2 { field: N } }]:`

error: aborting due to previous error

